app/javascript/controllers/forms_controller.js

```diff
@@ -1,17 +1,62 @@
-import { Controller } from "@hotwired/stimulus"
+import { Controller } from "@hotwired/stimulus";
 
 export default class extends Controller {
-    static targets = ["fileInput", "selectedFileName"];
+  static targets = [
+    "fileInput",
+    "selectedFileName",
+    "logoPreview",
+    "logoPreviewContainer",
+    "bannerPreview",
+    "bannerPreviewContainer"
+  ];
 
-    connect() {
-        this.fileInputTarget.addEventListener("change", (e) => {
-            const fileName = e.target.files[0]?.name || "No file chosen";
-            this.selectedFileNameTarget.innerText = fileName;
-        });
+  connect() {
+    this.fileInputTarget.addEventListener("change", (e) => {
+      const fileName = e.target.files[0];
+      this.updateFileName(fileName);
+      this.showPreview(fileName);
+    });
+  }
+
+  updateFileName(file) {
+    const fileName = file ? file.name : "No file chosen";
+    this.selectedFileNameTarget.innerText = fileName;
+  }
+
+  showPreview(file) {
+    const fileType = this.fileInputTarget.dataset.fileType;
```

Comment:
we don't really need to read the fileType here. There will be only one previewTarget either for banner or for logo. So just setting the src there should show the preview.

Reply:
This line of code removed